src/utils/requestNavigationHandlers.ts

```diff
@@ -158,8 +157,7 @@ export const handleMobileWalletProtocolRequest = async ({
       }
 
       if (action.method === 'wallet_switchEthereumChain') {
-        const chainId = BigNumber.from(action.params.chainId).toNumber();
-        const isSupportedChain = Object.values(ChainId).includes(chainId);
+        const isSupportedChain = SUPPORTED_MAINNET_CHAIN_IDS.includes(BigNumber.from(action.params.chainId).toNumber());
```

Comment:
same here - we might want to use `SUPPORTED_CHAIN_IDS` here as previously there wasn't a filter on testnets - iirc we wanted to be a bit more permissive in the initial connections